### DISCUSSION TITLE
Including activemq-cpp in a CMakeProject using VSCode

### DISCUSSION THREAD

**Discussion by Ruhrpottpatriot** [microsoft/vcpkg#41560] [Q&A]
I have a project which uses ActiveMQ to communicate between components. For the c++ based components I want to use activemq-cpp. However, when I include the library in my manifest file, neither the compiler, nor VSCode can find the required header files.

My manifest looks like this:
```
{
    "dependencies": [
        "activemq-cpp",
        "nlohmann-json"
    ]
}
```

while my CMakeLists.txt looks like this

```
cmake_minimum_required(VERSION 3.30)
cmake_policy(VERSION 3.30)

project(cb_tests
    LANGUAGES C CXX    
) 

find_package(activemq-cpp REQUIRED)
find_package(nlohmann_json CONFIG REQUIRED)

add_subdirectory(src)
add_subdirectory(include)
```

and my library CMakeLists.txt looks like this:
```
add_library(${CMAKE_PROJECT_NAME})

set_target_properties(${CMAKE_PROJECT_NAME}
    PROPERTIES 
        LINKER_LANGUAGE CXX 
        CXX_STANDARD 17
        POSITION_INDEPENDENT_CODE ON 
)

target_sources(${CMAKE_PROJECT_NAME}
    PRIVATE
        lib.cpp
    PUBLIC
        FILE_SET HEADERS
        BASE_DIRS
            ${CMAKE_SOURCE_DIR}/include
)
target_link_libraries(${CMAKE_PROJECT_NAME}
    PRIVATE
        nlohmann_json::nlohmann_json
        ${activemq-cpp_LIBRARIES}
)
```

and finally the culprit, lib.cpp:

```
#include "../include/cb/lib.h"
#include <activemq/core/ActiveMQConnectionFactory.h>   // <--- Can't find this file, even though it's present inside build/vcpkg_installed/x64-linux
#include <iostream>

Connector::Connector() {
    std::cout << "Created Connector!\n" << std::flush;
    this->msg_thread = std::thread(&Connector::message_sender, this);
}

Connector::~Connector() {
    std::cout << "====" << std::endl << "Connector Destructor" << std::endl << std::flush;

    this->is_running = false;
    if (this->msg_thread.joinable()) {
        this->msg_thread.join();
    }
}
```

As written in the code comments, neither the compiler, nor IntelliSense can find `activemq/core/ActiveMQConnectionFactory.h`. When I run CMake, it runs vcpkg install and installs activemq-cpp, but later it can't find it when the project is configured.

**Comment by Osyotr** [comment]
> ${activemq-cpp_LIBRARIES}

Should have been `unofficial::activemq-cpp::activemq-cpp`.
`activemq-cpp` does not have official cmake config hence `unofficial::` prefix.

**Reply by Ruhrpottpatriot** [reply]
Now it compiles fine, but VSCode IntelliSense still can't find the `#include` and it still doesn't show upin the "Found ..." messages.

**Comment by dg0yt** [comment]
LIBRARIES variables usually just carry linker information. If you use variables instead of import targets, you have to apply INCLUDE_DIRS variables as well.
